Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -422,3 +566,33 @@ def ReadFile(path):
         LOG_ACTION.log(LogType.Error, "Exception opening file " + path + " Error Code: " + str(error.errno) + " Error: " + error.message + error.strerror)
     return content
 
+def ConvertToUtf8(item):
+    return str(item).encode('utf-8')
+
+def GetWorkspaceListFromStateFile():
+    stateFileFullPath = os.path.join(PATH_NPM_STATE, FNAME_NPM_STATE)
+    fileContents = ReadFile(stateFileFullPath)
+    if fileContents is None:
+        return []
+    wsList = filter(IsUUID, fileContents.splitlines())
+    return map(ConvertToUtf8, wsList)
+
+def IsUUID(uuidStr):
+    try:
+        uuidOut = re.match('[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}\Z', uuidStr, re.I)
```

Comment:
Could not find the python uuid for python versions 2.4.6 hence went with the regex for this version. Exactly did what you meant in 2.6x and 3.x versions of the script. [2.6x Version of Script](https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/6b5fe772da38817c5215022f48f63043969309f8/Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAgentNPMConfig.py#L576)